File: BST/depthFirstSearch/preOrder.py
class TreeNode:
    def __init__(self,val):
        self.val = val
        self.left = None
        self.right = None

class Tree:
    def __init__(self,root):
        self.root = root
    
    def printTree(self):
        PrintPreOrder(self.root)

def PrintPreOrder(node):
    if node == None:
        return
    
    print(node.val)
    PrintPreOrder(node.left)
    PrintPreOrder(node.right)

if __name__ == "__main__":
  n1 = TreeNode(10)
  n2 = TreeNode(8)
  n3 = TreeNode(15)
  n4 = TreeNode(20)
  n1.left = n2
  n1.right = n3
  n3.right = n4
  t = Tree(n1)
  t.printTree()

File: BST/depthFirstSearch/preorder.js
function TreeNode(val){
    this.val = val;
    this.left = null;
    this.right = null;
  }
  
  // function BinarySearchTree(){
  //   this.root = null;
  // }

var p = new TreeNode(1);
p.left = new TreeNode(2);

p.right = new TreeNode(3);
p.left.left = new TreeNode(4);
p.left.right = new TreeNode(5);
p.right.left = new TreeNode(6);
p.right.right = new TreeNode(7);


var preorder = function(root){
    if(root == null){
      return
    } else {
      // console.log(root.val)
      preorder(root.left);
      preorder(root.right)
    }
  }
  
  preorder(p)